externs/browser/html5.js

```diff
@@ -4286,3 +4288,102 @@ Plugin.prototype.length;
 
 /** @type {string} */
 Plugin.prototype.name;
+
+/** @record */
+function CustomElementInterface() {}
+
+/** @type {!Array<string>|undefined} */
+CustomElementInterface.observedAttributes;
+
+/** @type {function()|undefined} */
+CustomElementInterface.prototype.connectedCallback;
+
+/** @type {function()|undefined} */
+CustomElementInterface.prototype.disconnectedCallback;
+
+/** @type {function(string, ?string, ?string, ?string)|undefined} */
+CustomElementInterface.prototype.attributeChangedCallback;
+
+/**
+ * @constructor
+ * @extends {HTMLElement}
+ * @implements {CustomElementInterface}
+ */
+function CustomElement () {}
+
+/** @type {!Array<string>|undefined} */
+CustomElement.observedAttributes;
+
+/** @type {function()|undefined} */
+CustomElement.prototype.connectedCallback;
+
+/** @type {function()|undefined} */
+CustomElement.prototype.disconnectedCallback;
+
+/** @type {function(string, ?string, ?string, ?string)|undefined} */
+CustomElement.prototype.attributeChangedCallback;
+
+/**
+ * @see https://w3c.github.io/webcomponents/spec/custom/#custom-elements-api
+ * @const
+ */
+var customElements = {};
+
+/**
+ * @param {string} tagName
+ * @param {!function(new:HTMLElement)} klass should also implement
+ *     the CustomElementInterface
+ * @param {{extends: string}=} options
+ * @return {!CustomElement}
+ */
+customElements.define = function (tagName, klass, options) {};
+
+/**
+ * @param {string} tagName
+ * @return {?CustomElement}
+ */
+customElements.get = function(tagName) {};
+
+/**
+ * @param {string} tagName
+ * @return {Promise<!CustomElement>}
+ */
+customElements.whenDefined = function(tagName) {};
+
+/** @type {boolean} */
+Event.prototype.composed;
+
+/**
+ * @return {!Array<!(Element|ShadowRoot|Document|Window)>}
+ */
+Event.prototype.composedPath = function() {};
+
+/**
+ * @param {!{mode: string}} options
```

Comment:
You can add an enum to "hint" that those are the only possible values, but it's not really enforced.